pkg/api/pod/util.go

```diff
@@ -260,3 +266,13 @@ func DropDisabledVolumeMountsAlphaFields(volumeMounts []api.VolumeMount) {
 		}
 	}
 }
+
+// DropDisabledVolumeDevicesAlphaFields removes disabled fields from []VolumeDevice.
+// This should be called from PrepareForCreate/PrepareForUpdate for all resources containing a VolumeDevice
+func DropDisabledVolumeDevicesAlphaFields(volumeDevices []api.VolumeDevice) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.BlockVolumeSupport) {
+		for i := range volumeDevices {
```

Comment:
Can we just set the entire slice to nil?

Reply:
ok, done